index.js

```diff
@@ -70,6 +73,30 @@ class Conf {
 	clear() {
 		this.store = obj();
 	}
+	onDidChange(key, callback) {
+		if (typeof key !== 'string') {
+			throw new TypeError(`Expected \`key\` to be of type \`string\`, got ${typeof key}`);
+		} else if (typeof callback !== 'function') {
+			throw new TypeError(`Expected \`callback\` to be of type \`function\`, got ${typeof callback}`);
+		}
+
+		let currentValue = key ? this.get(key) : this.store;
```

Comment:
I don't think it should fall back to `this.store`. Just do `let currentValue = this.get(key);` and the same with `newValue` further down.